File: frontend/note-app/src/types/Note.ts
export interface Note {
  id: number;
  title: string;
  content: string | null;
  createdAt: string; // ISO string representation of DateTime
  updatedAt: string; // ISO string representation of DateTime
}

export interface CreateNoteRequest {
  title: string;
  content?: string | null;
}

export interface UpdateNoteRequest {
  id: number;
  title: string;
  content?: string | null;
}

File: backend/NoteApi/Models/Note.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NoteApi.Models
{
    public class Note
    {
        public int Id { get; set; }
        
        [Required]
        public string Title { get; set; }
        
        public string? Content { get; set; }
        
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime CreatedAt { get; set; }
        
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime UpdatedAt { get; set; }
    }
}

